overlord/devicestate/handlers_remodel.go

```diff
@@ -115,13 +116,35 @@ func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) (err error) {
 		return err
 	}
 
+	var recoverySetup *recoverySystemSetup
 	if new.Grade() != asserts.ModelGradeUnset {
-		// TODO: update ubuntu-boot/device/model
-		// TODO: reseal for both the old and current model
+		var triedSystems []string
+		if err := st.Get("tried-systems", &triedSystems); err != nil {
+			return fmt.Errorf("cannot obtain tried recovery systems: %v", err)
+		}
+		recoverySetup, err = taskRecoverySystemSetup(t)
+		if err != nil {
+			return err
+		}
+		// should promoting or any of the later steps fails, the cleanup
+		// will be done in finalize-recovery-system undo
+		if err := boot.PromoteTriedRecoverySystem(remodCtx, recoverySetup.Label, triedSystems); err != nil {
+			return err
+		}
 	}
 
 	// and finish (this will set the new model)
-	return remodCtx.Finish()
+	if err := remodCtx.Finish(); err != nil {
+		return err
+	}
+
+	// TODO: consider reverting the model if reseal fails?
```

Comment:
it's unclear, we basically resealed with that model involved a bunch of times before getting here no? so is it a failure related to the model or it's a failure related to resealing, if we can't reseal we are probably in trouble either way